docs/sapling.md

```diff
@@ -0,0 +1,277 @@
+---
+title: Sapling Toolkit
+author: Roxane Letourneau
+---
+
+Sapling is a protocol allowing private transactions in a decentralized environment. 
+Sapling was introduced in Tezos in the Edo protocol. Refer to the Tezos documentation for more information on Sapling: https://tezos.gitlab.io/active/sapling.html
+
+## Keys
+
+**Spending key**
+
+The spending key is used to spend tokens. It must be handled securely to prevent funds from being lost or stolen.
+
+Taquito offers support for encrypted/unencrypted spending keys and mnemonics. Refer to the following link for more information: [InMemorySpendingKey](./sapling_in_memory_spending_key.md)
+
+**Viewing key**
+
+The viewing key is derived from the spending key. This key can be used to view all incoming and outgoing transactions. It must be handled securely to prevent a loss of privacy, as anyone accessing it can see the transaction history and the balance.
+
+Refer to the following link for more information: [InMemoryViewingKey](./sapling_in_memory_viewing_key.md)
+
+**Sapling address**
+
+Sapling addresses are used to receive tokens. They are derived from the viewing key.
+
+Here is an example on how to retrieve addresses: [InMemoryViewingKey](./sapling_in_memory_viewing_key.md#how-to-retrieve-payment-addresses-from-the-viewing-key)
+
+
+# Sapling toolkit
+
+The `@taquito/sapling` package provides a `SaplingToolkit` class that surfaces all of the Sapling capabilities, allowing it to read from a Sapling state and prepare transactions. 
+
+The constructor of the `SaplingToolkit` takes the following properties:
+- an instance of `InMemorySpendingKey` as the spending key is needed to prepare and sign transactions that spend tokens
+- the second parameter is an object containing:
+  - the address of the Sapling contract (string)
+  - the size of the memo of the corresponding Sapling contract (number)
+  - an optional Sapling id that must be specified if the contract contains more than one Sapling state.
+- an instance of a class implementing the `TzReadProvider` interface, which allows getting data from the blockchain
+- it is possible to specify a different packer than the `MichelCodecPacker`, which is used by default
+
+Here is an example of how to instantiate a `SaplingToolkit`:
+
+```ts
+import { TezosToolkit } from '@taquito/taquito';
+import { SaplingToolkit } from '@taquito/sapling';
+
+const tezos = new TezosToolkit('https://jakartanet.ecadinfra.com/');
+const saplingContract = await tezos.contract.at('KT1UYwMR6Q6LZnwQEi77DSBrAjKT1tEJb245');
+
+const inMemorySpendingKey = await InMemorySpendingKey.fromMnemonic('YOUR_MNEMONIC');
+
+const saplingToolkit = new SaplingToolkit(
+    inMemorySpendingKey, 
+    { contractAddress: saplingContract.address, memoSize: 8 }, 
+    tezos.getFactory(RpcReadAdapter)()
```

Comment:
It implements the TzReadProvider interface we introduced some time ago to facilitate indexers' integration with Taquito.
We could accept a RpcClient also if it is more convenient.

Reply:
It is a wrapper around the RpcClient, or a user can decide to pass a custom ReadProvider to read from an indexer instead of the RPC. I wanted to keep the same idea (interact with a ReadProvider rather than the RPC for more flexibility for the user).

The getFactory method from the TezosToolkit returns an instance of RpcReadAdapter directly, which is the easiest way to get it IMO. Otherwise, it require to deal with the "context" class which is not exposed to the users normally.

I am open to suggestions/opinions on this.

Reply:
I decided to refactor the RpcReadAdapter to avoid passing the context in the constructor: https://github.com/ecadlabs/taquito/pull/1802
And corrected the doc in this PR.